src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
+        Animal.nature();
+      }
+    }
```

Comment:
[CHECKLIST VIOLATION][CODE PERFORMANCE] - The method 'Animal.nature' is being called inside a loop (indirectly within the 'bite' method which can be called multiple times). This can lead to performance issues. It's better to manage the 'Animal.alive' array outside of the 'bite' method, for example, after all biting has occurred.